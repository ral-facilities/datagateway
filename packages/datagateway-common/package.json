{
  "name": "datagateway-common",
  "version": "0.1.0",
  "private": true,
  "files": [
    "lib"
  ],
  "main": "./lib/index.js",
  "dependencies": {
    "@date-io/date-fns": "^1.3.13",
    "@material-ui/pickers": "^3.2.10",
    "@material-ui/lab": "^4.0.0-alpha.57",
    "@types/lodash.debounce": "^4.0.6",
    "axios": "^0.21.1",
    "clsx": "^1.1.1",
    "connected-react-router": "^6.9.1",
    "date-fns": "^2.17.0",
    "i18next": "^19.9.0",
    "lodash.debounce": "^4.0.8",
    "loglevel": "^1.7.1",
    "react-draggable": "^4.4.3",
    "react-i18next": "^11.8.8",
    "react-redux": "^7.1.0",
    "react-router": "^5.0.1",
    "react-scripts": "4.0.3",
    "react-virtualized": "^9.22.3",
    "redux": "^4.0.4",
    "redux-mock-store": "^1.5.4",
    "redux-thunk": "^2.3.0"
  },
  "peerDependencies": {
    "@material-ui/core": ">=  4.0.0 < 5",
    "@material-ui/icons": ">=  4.0.0 < 5",
    "react": ">= 16.8.0 < 17",
    "react-dom": ">= 16.8.0 < 17",
    "react-router-dom": ">= 5.0.0 < 6"
  },
  "devDependencies": {
    "@material-ui/core": "^4.11.3",
    "@material-ui/icons": "^4.11.2",
    "@types/enzyme": "^3.10.8",
    "@types/jest": "^26.0.4",
    "@types/node": "^14.14.31",
    "@types/react": "^17.0.2",
    "@types/react-router-dom": "^5.1.7",
    "@types/react-virtualized": "^9.21.10",
    "@typescript-eslint/eslint-plugin": "^4.16.1",
    "@typescript-eslint/parser": "^4.16.1",
    "babel-eslint": "10.1.0",
    "enzyme": "^3.10.0",
    "enzyme-adapter-react-16": "^1.15.6",
    "enzyme-to-json": "^3.6.1",
    "eslint": "^7.5.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-config-react-app": "^6.0.0",
    "eslint-plugin-cypress": "^2.11.1",
    "eslint-plugin-flowtype": "^5.8.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsx-a11y": "^6.3.1",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-react": "^7.20.3",
    "eslint-plugin-react-hooks": "^4.1.0",
    "lint-staged": "^10.5.4",
    "prettier": "^2.2.1",
    "react": "^16.13.1",
    "react-dom": "^16.11.0",
    "react-router-dom": "^5.0.1",
    "typescript": "4.2.2",
    "tslib": "^2.1.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom --coverage --watchAll=false",
    "prepare": "yarn tsc",
    "test:watch": "react-scripts test --env=jsdom --watch",
    "lint:js": "eslint --ext=tsx --ext=ts --ext=js --ext=jsx --fix ./src",
    "tsc": "tsc -p tsconfig.lib.json",
    "eject": "react-scripts eject",
    "pre-commit": "lint-staged"
  },
  "lint-staged": {
    "src/**/*.{tsx,ts,js,jsx,json}": [
      "eslint --max-warnings=0 --ext=tsx --ext=ts --ext=js --ext=jsx  --fix",
      "prettier --write"
    ],
    "cypress/**/*.{tsx,ts,js,jsx,json}": [
      "eslint --max-warnings=0 --ext=tsx --ext=ts --ext=js --ext=jsx --fix",
      "prettier --write"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ]
  },
  "jest": {
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "collectCoverageFrom": [
      "src/**/*.{tsx,ts,js,jsx}",
      "!src/index.tsx",
      "!src/serviceWorker.ts",
      "!src/setupTests.js"
    ]
  }
}
