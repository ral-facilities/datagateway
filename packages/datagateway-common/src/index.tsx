// Uncomment to use App to test components
// import React from 'react';
// import ReactDOM from 'react-dom';
// import App from './App';
// import './index.css';

import { StateType } from './state/app.types';

export * from './app.types';
export { default as Preloader } from './preloader/preloader.component';
export * from './state/actions/actions.types';

export { default as Table } from './table/table.component';
export * from './table/table.component';
export { default as DetailsPanelRow } from './table/rowRenderers/detailsPanelRow.component';
export { default as DataHeader } from './table/headerRenderers/dataHeader.component';
export {
  default as TextColumnFilter,
  useTextFilter,
  usePrincipalExperimenterFilter,
} from './table/columnFilters/textColumnFilter.component';
export {
  default as DateColumnFilter,
  useDateFilter,
} from './table/columnFilters/dateColumnFilter.component';
export { default as ActionCellComponent } from './table/cellRenderers/actionCell.component';
export { default as DataCellComponent } from './table/cellRenderers/dataCell.component';
export { default as ExpandCellComponent } from './table/cellRenderers/expandCell.component';
export * from './table/cellRenderers/cellContentRenderers';

export { default as CardView } from './card/cardView.component';
export type { CardViewDetails } from './card/cardView.component';
export { default as AdvancedFilter } from './card/advancedFilter.component';

export * from './state/actions/index';
export * from './state/middleware/dgcommon.middleware';
export * from './state/app.types';
export * from './api/index';

export { default as DGCommonMiddleware } from './state/middleware/dgcommon.middleware';
export { default as createReducer } from './state/reducers/createReducer';
export {
  default as dGCommonReducer,
  initialState as dGCommonInitialState,
} from './state/reducers/dgcommon.reducer';

export type DGCommonState = StateType;

export * from './parseTokens';
export { default as handleICATError } from './handleICATError';
export { default as retryICATErrors } from './api/retryICATErrors';

export {
  default as ArrowTooltip,
  getTooltipText,
} from './arrowtooltip.component';
export { default as Sticky } from './sticky.component';
export { default as DGThemeProvider } from './dgThemeProvider.component';
export { default as Mark } from './mark.component';

export { default as HomePage } from './homePage/homePage.component';

export { default as AddToCartButton } from './views/addToCartButton.component';
export { default as ViewCartButton } from './views/viewCartButton.component';
export type { CartProps } from './views/viewCartButton.component';
export { default as ViewButton } from './views/viewButton.component';
export { default as ClearFiltersButton } from './views/clearFiltersButton.component';
export { default as DownloadButton } from './views/downloadButton.component';
export { default as SelectionAlert } from './views/selectionAlert.component';

export { default as ISISDatafileDetailsPanel } from './detailsPanels/isis/datafileDetailsPanel.component';
export { default as ISISDatasetDetailsPanel } from './detailsPanels/isis/datasetDetailsPanel.component';
export { default as ISISInstrumentDetailsPanel } from './detailsPanels/isis/instrumentDetailsPanel.component';
export { default as ISISInvestigationDetailsPanel } from './detailsPanels/isis/investigationDetailsPanel.component';
export { default as DLSDatafileDetailsPanel } from './detailsPanels/dls/datafileDetailsPanel.component';
export { default as DLSDatasetDetailsPanel } from './detailsPanels/dls/datasetDetailsPanel.component';
export { default as DLSVisitDetailsPanel } from './detailsPanels/dls/visitDetailsPanel.component';
export { default as InvestigationDetailsPanel } from './detailsPanels/investigationDetailsPanel.component';
export { default as DatasetDetailsPanel } from './detailsPanels/datasetDetailsPanel.component';
export { default as DatafileDetailsPanel } from './detailsPanels/datafileDetailsPanel.component';

export * from './urlBuilders';

// ReactDOM.render(<App />, document.getElementById('root'));
