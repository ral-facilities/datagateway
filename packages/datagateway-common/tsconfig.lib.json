{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "./lib",
    "target": "es5",
    "sourceMap": true,
    "rootDir": "src",
    "declaration": true,
    "declarationMap": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "importHelpers": true,
    "strictNullChecks": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": false,
    "lib": ["dom", "dom.iterable", "esnext"],
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "jsx": "react"
  },
  "exclude": [
    "lib",
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "src/setupTests.tsx",
    "**/*.test.*"
  ],
  "include": ["src"]
}
