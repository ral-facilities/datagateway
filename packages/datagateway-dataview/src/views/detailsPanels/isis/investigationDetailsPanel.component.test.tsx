import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import InvestigationsDetailsPanel from './investigationDetailsPanel.component';
import { Investigation } from 'datagateway-common';

describe('Investigation details panel component', () => {
  let shallow;
  let mount;
  let rowData: Investigation;
  const detailsPanelResize = jest.fn();
  const fetchDetails = jest.fn();
  const viewDatasets = jest.fn();

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'div' });
    mount = createMount();
    rowData = {
      id: 1,
      title: 'Test 1',
      name: 'Test 1',
      summary: 'foo bar',
      visitId: '1',
      rbNumber: '1',
      doi: 'doi 1',
      size: 1,
      investigationInstruments: [
        {
          id: 1,
          instrument: {
            id: 3,
            name: 'LARMOR',
          },
        },
      ],
      studyInvestigations: [
        {
          id: 11,
          study: {
            id: 12,
            pid: 'study pid',
          },
        },
      ],
      startDate: '2019-06-10',
      endDate: '2019-06-11',
    };
  });

  afterEach(() => {
    mount.cleanUp();
    detailsPanelResize.mockClear();
    fetchDetails.mockClear();
    viewDatasets.mockClear();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders user, sample and publication tabs when present in the data', () => {
    rowData.investigationUsers = [
      {
        id: 4,
        role: 'Investigator',
        user: {
          id: 5,
          name: 'Louise',
          fullName: 'Louise Davies',
        },
      },
      {
        id: 9,
        role: 'Investigator',
        user: {
          id: 10,
          name: 'Louise',
        },
      },
    ];

    rowData.samples = [
      {
        id: 7,
        name: 'Test sample',
      },
    ];

    rowData.publications = [
      {
        id: 8,
        fullReference: 'Test publication',
      },
    ];

    const wrapper = shallow(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls detailsPanelResize on load and when tabs are switched between', () => {
    rowData.publications = [
      {
        id: 8,
        fullReference: 'Test publication',
      },
    ];

    const wrapper = mount(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );

    expect(detailsPanelResize).toHaveBeenCalledTimes(1);

    wrapper
      .find('#investigation-publications-tab')
      .hostNodes()
      .simulate('click');

    expect(detailsPanelResize).toHaveBeenCalledTimes(2);
  });

  it('detailsPanelResize not called when not provided', () => {
    rowData.publications = [
      {
        id: 8,
        fullReference: 'Test publication',
      },
    ];

    const wrapper = mount(
      <InvestigationsDetailsPanel
        rowData={rowData}
        fetchDetails={fetchDetails}
      />
    );

    expect(detailsPanelResize).toHaveBeenCalledTimes(0);

    wrapper
      .find('#investigation-publications-tab')
      .hostNodes()
      .simulate('click');

    expect(detailsPanelResize).toHaveBeenCalledTimes(0);
  });

  it('calls fetchDetails on load if investigationUsers, samples or publications are missing', () => {
    mount(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );

    expect(fetchDetails).toHaveBeenCalledTimes(1);
    expect(fetchDetails).toHaveBeenCalledWith(1);
    fetchDetails.mockClear();

    rowData.investigationUsers = [];
    mount(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );

    expect(fetchDetails).toHaveBeenCalledTimes(1);
    expect(fetchDetails).toHaveBeenCalledWith(1);
    fetchDetails.mockClear();

    rowData.samples = [];
    mount(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );

    expect(fetchDetails).toHaveBeenCalledTimes(1);
    expect(fetchDetails).toHaveBeenCalledWith(1);
    fetchDetails.mockClear();

    rowData.publications = [];
    mount(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );
    expect(fetchDetails).not.toHaveBeenCalled();
  });

  it('gracefully handles StudyInvestigations without Studies and InvestigationUsers without Users', () => {
    rowData.studyInvestigations = [
      {
        id: 11,
      },
    ];

    rowData.investigationUsers = [
      {
        id: 4,
        role: 'Investigator',
      },
    ];

    const wrapper = shallow(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('push new address when clicking dataset tab', () => {
    const wrapper = mount(
      <InvestigationsDetailsPanel
        rowData={rowData}
        detailsPanelResize={detailsPanelResize}
        fetchDetails={fetchDetails}
        viewDatasets={viewDatasets}
      />
    );

    expect(viewDatasets).toHaveBeenCalledTimes(0);

    wrapper.find('#investigation-datasets-tab').hostNodes().simulate('click');

    expect(viewDatasets).toHaveBeenCalledTimes(1);
  });
});
