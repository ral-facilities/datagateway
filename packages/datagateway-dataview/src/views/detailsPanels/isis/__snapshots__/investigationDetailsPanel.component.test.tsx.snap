// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Investigation details panel component gracefully handles StudyInvestigations without Studies and InvestigationUsers without Users 1`] = `
<div
  id="details-panel"
  style={
    Object {
      "minWidth": 0,
    }
  }
>
  <WithStyles(ForwardRef(Tabs))
    aria-label="investigations.details.tabs_label"
    onChange={[Function]}
    scrollButtons="auto"
    value="details"
    variant="scrollable"
  >
    <WithStyles(ForwardRef(Tab))
      aria-controls="investigation-details-panel"
      id="investigation-details-tab"
      label="investigations.details.label"
      value="details"
    />
    <WithStyles(ForwardRef(Tab))
      aria-controls="investigation-users-panel"
      id="investigation-users-tab"
      label="investigations.details.users.label"
      value="users"
    />
    <WithStyles(ForwardRef(Tab))
      id="investigation-datasets-tab"
      label="investigations.details.datasets"
      onClick={[Function]}
    />
  </WithStyles(ForwardRef(Tabs))>
  <div
    aria-labelledby="investigation-details-tab"
    hidden={false}
    id="investigation-details-panel"
    role="tabpanel"
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h6"
        >
          <b>
            Test 1
          </b>
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Divider))
          className="makeStyles-divider-2"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.title
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            Test 1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.visit_id
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.summary
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            foo bar
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.doi
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            doi 1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.start_date
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            2019-06-10
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.end_date
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            2019-06-11
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
  <div
    aria-labelledby="investigation-users-tab"
    hidden={true}
    id="investigation-users-panel"
    role="tabpanel"
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1"
      container={true}
      direction="column"
    />
  </div>
</div>
`;

exports[`Investigation details panel component renders correctly 1`] = `
<div
  id="details-panel"
  style={
    Object {
      "minWidth": 0,
    }
  }
>
  <WithStyles(ForwardRef(Tabs))
    aria-label="investigations.details.tabs_label"
    onChange={[Function]}
    scrollButtons="auto"
    value="details"
    variant="scrollable"
  >
    <WithStyles(ForwardRef(Tab))
      aria-controls="investigation-details-panel"
      id="investigation-details-tab"
      label="investigations.details.label"
      value="details"
    />
    <WithStyles(ForwardRef(Tab))
      id="investigation-datasets-tab"
      label="investigations.details.datasets"
      onClick={[Function]}
    />
  </WithStyles(ForwardRef(Tabs))>
  <div
    aria-labelledby="investigation-details-tab"
    hidden={false}
    id="investigation-details-panel"
    role="tabpanel"
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h6"
        >
          <b>
            Test 1
          </b>
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Divider))
          className="makeStyles-divider-2"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.title
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            Test 1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.visit_id
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.summary
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            foo bar
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="11"
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.pid
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Link))
            href="https://doi.org/study pid"
          >
            study pid
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.doi
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            doi 1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.start_date
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            2019-06-10
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.end_date
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            2019-06-11
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`Investigation details panel component renders user, sample and publication tabs when present in the data 1`] = `
<div
  id="details-panel"
  style={
    Object {
      "minWidth": 0,
    }
  }
>
  <WithStyles(ForwardRef(Tabs))
    aria-label="investigations.details.tabs_label"
    onChange={[Function]}
    scrollButtons="auto"
    value="details"
    variant="scrollable"
  >
    <WithStyles(ForwardRef(Tab))
      aria-controls="investigation-details-panel"
      id="investigation-details-tab"
      label="investigations.details.label"
      value="details"
    />
    <WithStyles(ForwardRef(Tab))
      aria-controls="investigation-users-panel"
      id="investigation-users-tab"
      label="investigations.details.users.label"
      value="users"
    />
    <WithStyles(ForwardRef(Tab))
      aria-controls="investigation-samples-panel"
      id="investigation-samples-tab"
      label="investigations.details.samples.label"
      value="samples"
    />
    <WithStyles(ForwardRef(Tab))
      aria-controls="investigation-publications-panel"
      id="investigation-publications-tab"
      label="investigations.details.publications.label"
      value="publications"
    />
    <WithStyles(ForwardRef(Tab))
      id="investigation-datasets-tab"
      label="investigations.details.datasets"
      onClick={[Function]}
    />
  </WithStyles(ForwardRef(Tabs))>
  <div
    aria-labelledby="investigation-details-tab"
    hidden={false}
    id="investigation-details-panel"
    role="tabpanel"
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h6"
        >
          <b>
            Test 1
          </b>
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Divider))
          className="makeStyles-divider-2"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.title
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            Test 1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.visit_id
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.summary
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            foo bar
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="11"
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.pid
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Link))
            href="https://doi.org/study pid"
          >
            study pid
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.doi
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            doi 1
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.start_date
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            2019-06-10
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.end_date
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            2019-06-11
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
  <div
    aria-labelledby="investigation-users-tab"
    hidden={true}
    id="investigation-users-panel"
    role="tabpanel"
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="5"
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.users.name
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            Louise Davies
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="10"
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.users.name
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            Louise
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
  <div
    aria-labelledby="investigation-samples-tab"
    hidden={true}
    id="investigation-samples-panel"
    role="tabpanel"
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="7"
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.samples.name
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            Test sample
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
  <div
    aria-labelledby="investigation-publications-tab"
    hidden={true}
    id="investigation-publications-panel"
    role="tabpanel"
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="8"
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="overline"
        >
          investigations.details.publications.reference
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          <b>
            Test publication
          </b>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;
